hypothesis-python/src/hypothesis/strategies/_internal/recursive.py

```diff
@@ -29,8 +30,23 @@ class LimitedStrategy(SearchStrategy):
     def __init__(self, strategy):
         super().__init__()
         self.base_strategy = strategy
-        self.marker = 0
-        self.currently_capped = False
+        self.data = threading.local()
```

Comment:
The design looks good, but the name `data` by convention means "a `ConjectureData` object" - how about e.g. `self._threadlocal`?

Reply:
Yep, makes sense :) I found that naming in [the DynamicVariable implementation](https://github.com/HypothesisWorks/hypothesis/blob/845ec99d6d83bb3271b295e77c9d1a2e8a4fa4e0/hypothesis-python/src/hypothesis/utils/dynamicvariables.py#L23) (should we change the naming there?)